hydra-uber/src/main/resources/log4j2.conf

```diff
@@ -10,26 +8,31 @@ _status_min = ${?logging.status}
 name: hydra-oss-default, status: ${_status_min}, shutdownHook = disable
 appenders: {
   appender: [
-    { name: STDOUT,
-      type: Console,
-      PatternLayout: { pattern: "%p [%t] %d{ISO8601} %F (line %L) %.-5000m%n" },
-      target : SYSTEM_OUT
+    { name: query-trace,
+      type: RollingRandomAccessFile,
+      fileName: "${sys:hydra.logdir:-log}/query-trace/last",
+      filePattern: "${sys:hydra.logdir:-log}/query-trace/%d{yyyyMMdd-HHmmss}.gz",
+      PatternLayout: { pattern: "%p [%t] %d{ISO8601} %F (line %L) %m%n" },
+      Policies.Policies: [
+        { type: SizeBasedTriggeringPolicy, size: 250 MB },
+        { type: OnStartupTriggeringPolicy }
+      ]
     },
-    { name: STDERR,
-      type: Console,
+    { name: File,
+      type: RollingRandomAccessFile,
+      fileName: "${sys:hydra.logdir:-log}/${sys:logfile.name}.log",
+      filePattern: "${sys:hydra.logdir:-log}/old/${sys:logfile.name}-%d{yyMMddHHmmss}.log.gz",
       PatternLayout: { pattern: "%p [%t] %d{ISO8601} %F (line %L) %.-5000m%n" },
-      target : SYSTEM_ERR
+      OnStartupTriggeringPolicy: { minSize: 1 }
```

Comment:
Let's discuss if we should use daily rolling instead of every startup.

Reply:
Decided to use daily rolling and set the period to 86400 seconds which is 24 hours. Log4j will find the smallest unit in filePattern and use that unit as the unit for the parameter in the TimeTriggeringPolicy. Since the smallest unit I used in filePattern is ss (seconds) that means I need to put 86400 to represent 1 day.